openverse_catalog/dags/providers/provider_csv_load_scripts/inaturalist/05_export_to_json_template.sql

```diff
@@ -0,0 +1,93 @@
+  sort / index, but might have introduced data quality risks (if postgres moved
+  things around while the job was running) and some pages appeared empty which required
+  more complicated python logic for retries. More on this approach at:
+  https://www.citusdata.com/blog/2016/03/30/five-ways-to-paginate/
+
+Everything on iNaturalist is holding at version 4, except CC0 which is version 1.0.
+License versions below are hard-coded from inaturalist
+https://github.com/inaturalist/inaturalist/blob/d338ba76d82af83d8ad0107563015364a101568c/app/models/shared/license_module.rb#L5
+*/
+
+SELECT
+    json_build_object(
+        'foreign_id',
+        INATURALIST.PHOTOS.PHOTO_ID,
+        'width',
+        INATURALIST.PHOTOS.WIDTH,
+        'height',
+        INATURALIST.PHOTOS.HEIGHT,
+        'title',
+        INATURALIST.TAXA.NAME,
+        'tags',
+        INATURALIST.TAXA.TAGS,
+        'filetype',
+        LOWER(INATURALIST.PHOTOS.EXTENSION),
+        'license_url',
+        (CASE
+            WHEN INATURALIST.PHOTOS.LICENSE = 'CC-BY-NC-SA'
+                THEN 'http://creativecommons.org/licenses/by-nc-sa/4.0/'
+            WHEN INATURALIST.PHOTOS.LICENSE = 'CC-BY-NC'
+                THEN 'http://creativecommons.org/licenses/by-nc/4.0/'
+            WHEN INATURALIST.PHOTOS.LICENSE = 'CC-BY-NC-ND'
+                THEN 'http://creativecommons.org/licenses/by-nc-nd/4.0/'
+            WHEN INATURALIST.PHOTOS.LICENSE = 'CC-BY'
+                THEN 'http://creativecommons.org/licenses/by/4.0/'
+            WHEN INATURALIST.PHOTOS.LICENSE = 'CC-BY-SA'
+                THEN 'http://creativecommons.org/licenses/by-sa/4.0/'
+            WHEN INATURALIST.PHOTOS.LICENSE = 'CC-BY-ND'
+                THEN 'http://creativecommons.org/licenses/by-nd/4.0/'
+            WHEN INATURALIST.PHOTOS.LICENSE = 'PD'
+                THEN 'http://en.wikipedia.org/wiki/Public_domain'
+            WHEN INATURALIST.PHOTOS.LICENSE = 'CC0'
+                THEN 'http://creativecommons.org/publicdomain/zero/1.0/'
+            END),
+        'foreign_landing_url',
+        'https://www.inaturalist.org/photos/' || INATURALIST.PHOTOS.PHOTO_ID,
+        'image_url',
+        'https://inaturalist-open-data.s3.amazonaws.com/photos/'
+        || INATURALIST.PHOTOS.PHOTO_ID || '/medium.' || INATURALIST.PHOTOS.EXTENSION,
+        'creator',
+        COALESCE(INATURALIST.OBSERVERS.LOGIN, INATURALIST.PHOTOS.OBSERVER_ID::text),
+        'creator_url',
+        'https://www.inaturalist.org/users/' || INATURALIST.PHOTOS.OBSERVER_ID
+    )
+FROM INATURALIST.PHOTOS
+INNER JOIN
+    INATURALIST.OBSERVATIONS ON
+        INATURALIST.PHOTOS.OBSERVATION_UUID = INATURALIST.OBSERVATIONS.OBSERVATION_UUID
+INNER JOIN
+    INATURALIST.OBSERVERS ON
+        INATURALIST.PHOTOS.OBSERVER_ID = INATURALIST.OBSERVERS.OBSERVER_ID
+INNER JOIN
+    INATURALIST.TAXA ON
+        INATURALIST.OBSERVATIONS.TAXON_ID = INATURALIST.TAXA.TAXON_ID
+ORDER BY PHOTO_ID
+LIMIT batch_limit
+OFFSET offset_num ;
```

Comment:
Maybe we could make the operation of this script clearer by turning them into Python `str::format` templates?

```suggestion
LIMIT {batch_limit}
OFFSET {offset_num};
```

Then rather than the `.replace` call in the provider class, it's a `.format` call with the appropriate kwargs. To further help, we can change the filename slightly to clarify the intention:

```diff
- 05_export_to_json_template.sql
+ 05_export_to_json.template.sql
```

Reply:
Yeah, that's much nicer. Thanks!

Reply:
Great idea! One thing to note is that if we have brackets anywhere in the file, python will try interpolating those as well and may cause errors. I ran into that with #608 and had to use `%(value)s` formatting instead: https://github.com/WordPress/openverse-catalog/pull/608/files#diff-466e53b939b93d92ed5f2265131c3b747e4e1402ac94276b83e3de562576b99e